TophatModules/Sources/TophatControlServices/TophatRemoteControlRequest.swift

```diff
@@ -0,0 +1,26 @@
+//
+//  TophatRemoteControlRequest.swift
+//  TophatControlServices
+//
+//  Created by Lukas Romsicki on 2024-12-02.
+//  Copyright © 2024 Shopify. All rights reserved.
+//
+
+import Foundation
+
+/// The protocol you use to define notifications to be sent between Tophat processes.
+public protocol TophatRemoteControlRequest: Codable {
+	associatedtype Reply: Codable
+
+	var id: UUID { get }
+}
+
+extension TophatRemoteControlRequest {
+	var notificationName: String {
+		String(describing: type(of: self))
```

Comment:
I wonder if it's worth adding the bundle identifier as a `notificationName` prefix? I suppose these type names are already pretty unique.